src/openrct2/world/Fountain.cpp

```diff
@@ -65,150 +65,140 @@ const uint8_t _fountainPatternFlags[] = {
+                                                                      : SPRITE_MISC_JUMPING_FOUNTAIN_WATER;
+        jumpingFountain->NumTicksAlive = 0;
         jumpingFountain->frame = 0;
     }
 }
 
-void jumping_fountain_update(rct_jumping_fountain* jumpingFountain)
+void JumpingFountain::Update()
 {
-    jumpingFountain->num_ticks_alive++;
-    // Originaly this would not update the frame on the following
+    NumTicksAlive++;
+    // Originally this would not update the frame on the following
     // ticks: 1, 3, 6, 9, 11, 14, 17, 19, 22, 25
-    // This change was to simplefy the code base. There is a small increase
+    // This change was to simplify the code base. There is a small increase
     // in speed of the fountain jump because of this change.
-    if ((jumpingFountain->num_ticks_alive % 3) == 0)
+    if (NumTicksAlive % 3 == 0)
     {
         return;
     }
 
-    invalidate_sprite_0((rct_sprite*)jumpingFountain);
-    jumpingFountain->frame++;
+    invalidate_sprite_0(reinterpret_cast<rct_sprite*>(this));
+    frame++;
 
-    switch (jumpingFountain->type)
+    switch (type)
     {
         case SPRITE_MISC_JUMPING_FOUNTAIN_WATER:
-            if (jumpingFountain->frame == 11 && (jumpingFountain->fountain_flags & FOUNTAIN_FLAG::FAST))
+            if (frame == 11 && (FountainFlags & FOUNTAIN_FLAG::FAST))
             {
-                jumping_fountain_continue(jumpingFountain);
+                Continue();
             }
-            if (jumpingFountain->frame == 16 && !(jumpingFountain->fountain_flags & FOUNTAIN_FLAG::FAST))
+            if (frame == 16 && !(FountainFlags & FOUNTAIN_FLAG::FAST))
             {
-                jumping_fountain_continue(jumpingFountain);
+                Continue();
             }
             break;
         case SPRITE_MISC_JUMPING_FOUNTAIN_SNOW:
-            if (jumpingFountain->frame == 16)
+            if (frame == 16)
             {
-                jumping_fountain_continue(jumpingFountain);
+                Continue();
             }
             break;
     }
 
-    if (jumpingFountain->frame == 16)
+    if (frame == 16)
     {
-        sprite_remove((rct_sprite*)jumpingFountain);
+        sprite_remove(reinterpret_cast<rct_sprite*>(this));
     }
 }
 
-static int32_t jumping_fountain_get_type(const rct_jumping_fountain* jumpingFountain)
+int32_t JumpingFountain::GetType() const
 {
-    int32_t type = jumpingFountain->type == SPRITE_MISC_JUMPING_FOUNTAIN_SNOW ? JUMPING_FOUNTAIN_TYPE_SNOW
-                                                                              : JUMPING_FOUNTAIN_TYPE_WATER;
-    return type;
+    const int32_t fountainType = type == SPRITE_MISC_JUMPING_FOUNTAIN_SNOW ? JUMPING_FOUNTAIN_TYPE_SNOW
+                                                                           : JUMPING_FOUNTAIN_TYPE_WATER;
+    return fountainType;
 }
 
-static void jumping_fountain_continue(rct_jumping_fountain* jumpingFountain)
+void JumpingFountain::Continue()
```

Comment:
I think a better name for this would be maybe ```Advance``` given that ```continue``` is a reserved keyword.

Reply:
How about AdvanceAnimation or ContinueAnimation to be consistent with the change to Begin -> BeginAnimation?

Reply:
Yeah true